extensions/email/src/main/java/org/mule/extension/email/internal/commands/ListCommand.java

```diff
@@ -41,32 +43,38 @@
      * For folder implementations (like IMAP) that support fetching without reading the content, if the content
      * should NOT be read ({@code readContent} = false) the SEEN flag is not going to be set.
      *
-     * @param connection  the associated {@link RetrieverConnection}.
-     * @param folderName  the name of the folder where the emails are stored.
-     * @param readContent if should read the email content or not.
-     * @param matcher     a {@link Predicate} of {@link EmailAttributes} used to filter the output list  @return a {@link List} of {@link MuleMessage} carrying all the emails and it's corresponding attributes.
+     * @param connection      the associated {@link RetrieverConnection}.
+     * @param folderName      the name of the folder where the emails are stored.
+     * @param contentEncoding the encoding to be used while reading the emails content
+     * @param readContent     if should read the email content or not.
+     * @param matcher         a {@link Predicate} of {@link EmailAttributes} used to filter the output list  @return a {@link List} of {@link MuleMessage} carrying all the emails and it's corresponding attributes.
      */
-    public List<MuleMessage> list(RetrieverConnection connection,
-                                  String folderName,
-                                  boolean readContent,
-                                  Predicate<EmailAttributes> matcher)
+    public List<Email> list(RetrieverConnection connection,
+                            String folderName,
+                            String contentEncoding,
```

Comment:
not necesarilly. what we usually do is to allow operations to use a custom encoding. If not provided then we go to the config. If still not provided, default to mule's default.

Check file and sockets for examples